### PR TITLE
locations mapping

### PR DESCRIPTION
**Pull request by banukutlu**
#121

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ Style/MixinUsage:
   Exclude:
     - 'lib/traject/psulib_config.rb'
 
-# Offense count: 57
+# Offense count: 59
```
Comment (cdmo) [inline .rubocop_todo.yml:60]: might want to add an exclude for `setting.yml` or turn that large array into a format like 

```yml
location_excludes:
  - item 1
  - item 2
...
```

small thing

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -35,7 +35,7 @@ Metrics/MethodLength:
 # Offense count: 1
 # Configuration parameters: CountComments, CountAsOne.
 Metrics/ModuleLength:
-  Max: 174
+  Max: 180
```
Comment (cdmo) [inline .rubocop_todo.yml:38]: Wonder if we shouldn't add an exclude here for `psulib_config.rb` 🤔

File: spec/lib/traject/macros/custom_spec.rb [hunk 3]
```diff
@@ -263,4 +263,15 @@
       end
     end
   end
+
+  describe '#exclude_locations' do
+    let(:location_1) { { '949' => { 'ind1' => '', 'ind2' => '', 'subfields' => [{ 'l' => 'DOCUSMF-DN' }] } } }
+    let(:location_2) { { '949' => { 'ind1' => '', 'ind2' => '', 'subfields' => [{ 'l' => 'RESERVE-HN' }] } } }
+    let(:location_3) { { '949' => { 'ind1' => '', 'ind2' => '', 'subfields' => [{ 'l' => 'AVAIL_SOON' }] } } }
+    let(:result) { @indexer.map_record(MARC::Record.new_from_hash('fields' => [location_1, location_2, location_3], 'leader' => leader)) }
```
Comment (cdmo) [inline spec/lib/traject/macros/custom_spec.rb:271]: latest version of rubocop seem to wanting to use let memoization less. 🤷‍♂️ It does feel easier to read and it is shorter.

```suggestion
    location_1 = { '949' => { 'ind1' => '', 'ind2' => '', 'subfields' => [{ 'l' => 'DOCUSMF-DN' }] } }
    location_2 = { '949' => { 'ind1' => '', 'ind2' => '', 'subfields' => [{ 'l' => 'RESERVE-HN' }] } }
    location_3 = { '949' => { 'ind1' => '', 'ind2' => '', 'subfields' => [{ 'l' => 'AVAIL_SOON' }] } }
    result = @indexer.map_record(MARC::Record.new_from_hash('fields' => [location_1, location_2, location_3], 'leader' => leader))
```

File: lib/translation_maps/locations.properties [hunk 4]
```diff
@@ -0,0 +1,709 @@
+ACAD-BLDG = Submit request for annexed material
```
Comment (cdmo) [inline lib/translation_maps/locations.properties:1]: We can't get this from a web service call, right?

File: lib/traject/macros/custom.rb [hunk 5]
```diff
@@ -226,6 +226,13 @@ def hathi_to_hash(ht_format)
         end
       end
 
+      def exclude_locations
+        lambda do |_record, accumulator|
+          accumulator.reject! { |value| ConfigSettings.location_excludes.include?(value) }
+          accumulator.reject!(&:empty?)
```
Comment (cdmo) [inline lib/traject/macros/custom.rb:232]: is this the same as `accumulator.compact`? Just eliminating empty items?

**Review by cdmo — CHANGES_REQUESTED** [review]
Couple of question/thoughts, looks good though.

File: lib/traject/macros/custom.rb (hunk 5 above)
Comment (banukutlu) [reply lib/traject/macros/custom.rb:232]: ah yes most likely, I just forget about that method :) I will check and change it so it is consistent with the rest

Comment (cdmo) [reply lib/traject/macros/custom.rb:232]: I do love me a pretzel colon operator - but yeah I think if it's the same as compact that probably makes more sense (being nitpicky I know sorry)

File: .rubocop_todo.yml (hunk 2 above)
Comment (banukutlu) [reply .rubocop_todo.yml:38]: 180 is from `lib/traject/macros/custom.rb`

File: lib/translation_maps/locations.properties (hunk 4 above)
Comment (banukutlu) [reply lib/translation_maps/locations.properties:1]: I am not sure honestly, can check with Maryam

File: .rubocop_todo.yml (hunk 2 above)
Comment (cdmo) [reply .rubocop_todo.yml:38]: oh right!

File: lib/translation_maps/locations.properties (hunk 4 above)
Comment (cdmo) [reply lib/translation_maps/locations.properties:1]: ok, if she says yes we can follow up with a new PR to replace 👍

**Review by cdmo — APPROVED** [review]

File: lib/translation_maps/locations.properties (hunk 4 above)
Comment (banukutlu) [reply lib/translation_maps/locations.properties:1]: exactly what I was thinking 👍 :)

**Pull request merged**